### PR TITLE
split database CA into server/client CAs

### PR DESCRIPTION
**Pull request by GavinFrazar**
Fixes https://github.com/gravitational/teleport-private/issues/782

changelog: Introduced the Teleport Database Client CA so that separate CAs are used to issue certs to databases and database clients.

### CODE REVIEW

File: lib/auth/grpcserver.go [hunk 1]
```diff
@@ -475,6 +479,127 @@ func WatchEvents(watch *authpb.Watch, stream WatchEvent, componentName string, a
...
+func getClientVersion(ctx context.Context) (*semver.Version, error) {
+	clientVersionString, ok := metadata.ClientVersionFromContext(ctx)
+	if !ok {
+		return nil, trace.NotFound("no client version found in grpc context")
+	}
+	clientVersion, err := semver.NewVersion(clientVersionString)
+	return clientVersion, trace.Wrap(err)
+}
+
+// versionSupportsDatabaseClientCA returns true if the client version supports
```
Comment (greedy52) [inline lib/auth/grpcserver.go:567]: nit: might want to update the function name or description as it's checking the version with CA filter

**Review by greedy52 — APPROVED** [review]

File: lib/auth/grpcserver.go (hunk 1 above)
Comment (GavinFrazar) [reply lib/auth/grpcserver.go:567]: yeah good point, it's really more like "versionUsesCAFilters"

Although now I realize there's not really a need for version checking at all - if it's unfiltered then it must be a version that pre-dates the CA filtering changes (and the CA split), so we can just inject a filter if there isn't one.

Comment (greedy52) [reply lib/auth/grpcserver.go:567]: true 😀

Comment (GavinFrazar) [reply lib/auth/grpcserver.go:567]: eh, to be on the safe side I'll keep the version checking, it might prevent an issue in the future

Comment (GavinFrazar) [reply lib/auth/grpcserver.go]: [8634ef7](https://github.com/gravitational/teleport/pull/35949/commits/8634ef7da2707fc0143616d54613b3f28ef7868f)

**Review by smallinsky — APPROVED** [review]

**Pull request merged**